### PR TITLE
Refactor Chinese GetCustomNumberModel

### PR DESCRIPTION
**Pull request by JuanAr**
This updates are based on the changes made to the `test-refactor` branch since tests is where the GetCustomNumberModel  methods where originally placed. This PR should be merged only after the tests refactor are merged into `master`.

In this PR the logic to create the Custom Chinese Number Model we move into the Recognizer + Configuration + Extractors _pipeline_.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs [hunk 1]
```diff
@@ -6,5 +6,6 @@ namespace Microsoft.Recognizers.Text.Number
     public enum NumberOptions
     {
         None = 0,
+        ChineseNumbersWithPercent = 1
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs]: I don't think this is something we should surface at the recognizer level. This is more an implementation detail of one of the Chinese number models. 
We'll do a closer analysis of the change and update the PR review soon.

File: JavaScript/test/runner-number.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ var modelFunctions = {
     'PercentModel': (input, culture, options) => Recognizer.recognizePercentage(input, culture, options, false),
     // TODO: Implement number range model in javascript
     'NumberRangeModel': (input, culture, options) => null,
-    'CustomNumberModel': (input, culture, options) => getCustomNumberModel(culture).parse(input)
```
Comment (tellarin) [inline JavaScript/test/runner-number.js]: If custom model was only used to test this particular case, I think we can remove it.
The tests for the Chinese Units model will already make use of it.

File: .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs (hunk 1 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs]: Changes in PR https://github.com/Microsoft/Recognizers-Text/pull/379

Comment (JuanAr) [reply .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs]: ACK. Please, let me know if we should solve conflicts for this PR or Close it without merging.

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.Number/NumberOptions.cs]: Let's close this one.
I'll remove the custom model test in another PR as it only target one case, and it's tested by the unit model tests anyways. It's easier that way.

**Comment by tellarin** [comment]
Closing as discussed. Custom test will be removed as it doesn't add much value.

**Pull request closed without merging**